File: src/main/java/com/serfagab/entities/OrdenCompra.java
package com.serfagab.entities;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.List;

@Data
@Entity
@Table(name = "tbl_orden_compra")
public class OrdenCompra {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_orden_compra")
    private Integer idOrdenCompra;

    @ManyToOne
    @JoinColumn(name = "id_proveedor")
    private Proveedor proveedor;

    @ManyToOne
    @JoinColumn(name = "id_usuario")
    private Usuario usuario;

    @Column(name = "fecha")
    private LocalDate fecha;

    @Column(name = "estado")
    private String estado = "PENDIENTE";

    @Column(name = "total")
    private Double total = 0.0;

    @Column(name = "observaciones")
    private String observaciones;

    @OneToMany(mappedBy = "ordenCompra", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<DetalleOrdenCompra> detalles;
}

File: frontend-angular/src/app/models/orden-compra.ts
import { Proveedor } from './proveedor';
import { Usuario } from './usuario';
import { DetalleOrden } from './detalle-orden';

export interface OrdenCompra {
  idOrdenCompra: number;
  proveedor: Proveedor;
  usuario: Usuario;
  fecha: string;
  estado: string;
  total: number;
  observaciones: string;
  detalles: DetalleOrden[];
}